src/main/kotlin/com/vaadin/gradle/ReflectionsClassFinder.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.gradle
+
+import com.vaadin.flow.server.frontend.scanner.ClassFinder
+import org.reflections.Reflections
+import org.reflections.util.ConfigurationBuilder
+import java.lang.annotation.Repeatable
+import java.lang.reflect.AnnotatedElement
+import java.net.URL
+import java.net.URLClassLoader
+
+/**
+ * A class finder using org.reflections.
+ *
+ * @param urls
+ *            the list of urls for finding classes.
+ *
+ * @since 2.0
+ */
+class ReflectionsClassFinder(vararg urls: URL?) : ClassFinder {
```

Comment:
@mvysny 
Since ReflectionsClassFinder.java: [1] has been moved from flow-migration:2.2 to flow-maven-plugin:3.1, we created the ReflectionsClassFinder in vaadin-gradle-plugin, it's the same code but in Kotlin 
[1]: https://github.com/vaadin/flow/blob/3.1/flow-maven-plugin/src/main/java/com/vaadin/flow/server/scanner/ReflectionsClassFinder.java

Reply:
Maybe flow-maven-plugin could be added as a dependency and we could use that helper directly from Gradle plugin? We have discussed that earlier with @mvysny, but never really tried that.

Reply:
That could be possible, but it would pull in a lot of Maven-based dependencies, and I'm not sure whether we want to have that. Ideally there should be a jar file that the Gradle plugin could use, and which would contain all such helper classes. Like flow-server :)

Reply:
I wouldn't be afraid of non-used dependencies unless there are conflicts. Lot of things can probably be excluded as well. The correct way would of course be to extract common tools to a separate module, but you know, that might take a while...

Reply:
@mstahv @mvysny  Maybe we can use flow-maven-plugin as a dependency and exclude non-used transitive dependencies.
What do you think?